zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/ClientTimeoutHandler.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.zuul.netty.connectionpool;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelOutboundHandlerAdapter;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.CombinedChannelDuplexHandler;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.util.AttributeKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Client Timeout Handler
+ *
+ * Author: Arthur Gonigberg
+ * Date: July 01, 2019
+ */
+public class ClientTimeoutHandler extends CombinedChannelDuplexHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(ClientTimeoutHandler.class);
+
+    public static final AttributeKey<Integer> ORIGIN_RESPONSE_READ_TIMEOUT = AttributeKey.newInstance("originResponseReadTimeout");
+
+    public ClientTimeoutHandler() {
+        super(new InboundHandler(), new OutboundHandler());
+    }
+
+    private static class InboundHandler extends ChannelInboundHandlerAdapter {
```

Comment:
I -think- we can instead extend SimpleChannelInboundHandler<LastHttpContent> to remove the `msg instanceof LastHttpContent` check and not worry about reference counting.